src/ride/serializers.py

```diff
@@ -0,0 +1,65 @@
+from django.contrib.auth.models import User
+from path.serializers import PathSerializer
+from person.serializers import UserSerializer
+from rest_framework import serializers
+from ride.models import Ride
+
+
+class RideSerializer(serializers.ModelSerializer):
+    id = serializers.IntegerField(source="path.id")
+    creator = serializers.SerializerMethodField("get_creator")
+    driver = serializers.SerializerMethodField("get_driver")
+    riders = serializers.SerializerMethodField("get_riders")
+    path = serializers.SerializerMethodField("get_path")
+
+    class Meta:
+        model = Ride
+        fields = (
+            "id",
+            "creator",
+            "max_travelers",
+            "min_travelers",
+            "departure_datetime",
+            "description",
+            "driver",
+            "is_flexible",
+            "riders",
+            "estimated_cost",
+            "path",
+            "type",
+        )
+        read_only_fields = fields
+
+    def get_creator(self, ride):
+        if not ride.creator:
+            return None
+        if not User.objects.filter(id=ride.creator.id).exists():
+            return None
+        creator = UserSerializer(User.objects.get(id=ride.creator.id)).data
+        if creator == {}:
+            return None
+        return creator
+
+    def get_driver(self, ride):
+        if not ride.driver:
+            return None
+        if User.objects.filter(id=ride.driver.id).exists():
+            return None
+        driver = UserSerializer(User.objects.get(id=ride.driver.id)).data
+        if driver == {}:
+            return None
+        return driver
```

Comment:
all of these methods are very inefficient
you want to do as minimal computation in serializers as possible
because serializers make a trip to the database for every object, which doesn't let django optimize queries (take less trips to the db)

so try not to do a lot of stuff like this

Reply:
if you can calculate stuff, i'd recommend setting properties in the ride model so that you can do ride.get_riders or something like that, or simply just do the serializer thing i commented about before